src/lib/fixer.ts

```diff
@@ -131,6 +138,164 @@ export class _PmdFixGenerator extends FixGenerator {
             title: 'Clear Single Diagnostic',
             arguments: [this.document.uri, this.diagnostic]
         };
+
         return action;
     }
+
+    public generateClassLevelSuppression(): vscode.CodeAction {
+        // Find the end-of-line position of the class declaration where the diagnostic is found.
+        const classStartPosition = this.findClassStartPosition(this.diagnostic, this.document);
+        // const classEndOfLinePosition = 0;
+
+        const action = new vscode.CodeAction(messages.fixer.supressOnClass, vscode.CodeActionKind.QuickFix);
+        action.edit = new vscode.WorkspaceEdit();
+    
+        // Determine the appropriate suppression rule based on the type of diagnostic.code
+        let suppressionRule: string;
+        if (typeof this.diagnostic.code === 'object' && this.diagnostic.code !== null && 'value' in this.diagnostic.code) {
+            suppressionRule = `PMD.${this.diagnostic.code.value}`;
+        } else {
+            suppressionRule = `PMD`;
+        }
+    
+        // Extract text from the start to end of the class declaration to search for existing suppressions
+        const classText = this.findLineBeforeClassStartDeclaration(classStartPosition, this.document);
+        const suppressionMatch = classText.match(this.suppressionRegex);
+    
+        if (suppressionMatch) {
+            // If @SuppressWarnings exists, check if the rule is already present
+            const existingRules = suppressionMatch[1].split(',').map(rule => rule.trim());
+            if (!existingRules.includes(suppressionRule)) {
+                // If the rule is not present, add it to the existing @SuppressWarnings
+                const updatedRules = [...existingRules, suppressionRule].join(', ');
+                const updatedSuppression = this.generateUpdatedSuppressionTag(updatedRules, this.document.languageId);
+                const suppressionStartPosition = this.document.positionAt(classText.indexOf(suppressionMatch[0]));
+                const suppressionEndPosition = this.document.positionAt(classText.indexOf(suppressionMatch[0]) + suppressionMatch[0].length);
+                const suppressionRange = new vscode.Range(suppressionStartPosition, suppressionEndPosition);
+                action.edit.replace(this.document.uri, suppressionRange, updatedSuppression);
+            }
+        } else {
+            // If @SuppressWarnings does not exist, insert a new one
+            const newSuppression = this.generateNewSuppressionTag(suppressionRule, this.document.languageId);
+            action.edit.insert(this.document.uri, classStartPosition, newSuppression);
+        }
+    
+        action.diagnostics = [this.diagnostic];
+        action.command = {
+            command: Constants.COMMAND_REMOVE_DIAGNOSTICS_ON_SELECTED_FILE,
```

Comment:
FYI, I clear the diagnostics for the whole file and this is not ideal when a file has multiple classes. Keeping track of the diagnostics that would be affected by the rule added on the specific class seemed too difficult and the AC for the story did not call for it. Since this PR is already doing too many things, I could still do it as part of a separate story.

Reply:
Clarification: Does "multiple classes" mean multiple outer classes, or an outer class with inner classes? I know Apex can't have multiple outer classes, and I'm pretty sure that the annotation will only apply to outer classes, though I haven't checked that.

Reply:
Didn't realize apex can't have multiple outer classes. I have changed the code and now this applies only to the outer class.